tests/unit/dataactvalidator/test_d41_detached_award_financial_assistance_5.py

```diff
@@ -0,0 +1,78 @@
+    det_award_2 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="Ny**123",
+                                                          place_of_performance_zip4a=None)
+    # valid 5 digit zip
+    det_award_3 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="Ny**123",
+                                                          place_of_performance_zip4a="12345")
+    det_award_4 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="NY98765",
+                                                          place_of_performance_zip4a="12345")
+    # valid 9 digit zip
+    det_award_5 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="NY98765",
+                                                          place_of_performance_zip4a="123456789")
+    det_award_6 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny98765",
+                                                          place_of_performance_zip4a="123456789")
+    det_award_7 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny98765",
+                                                          place_of_performance_zip4a="12345-6789")
+    errors = number_of_errors(_FILE, database, models=[det_award_1, det_award_2, det_award_3, det_award_4, det_award_5,
+                                                       det_award_6, det_award_7, zips])
+    assert errors == 0
+
+    # random wrong length zips and zips with '-' in the wrong place, formatting is checked in another rule
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny10986",
+                                                          place_of_performance_zip4a="12345678")
+    det_award_2 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny10986",
+                                                          place_of_performance_zip4a="1234567898")
+    det_award_3 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny10986",
+                                                          place_of_performance_zip4a="12345678-9")
+    det_award_4 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny10986",
+                                                          place_of_performance_zip4a="123-456789")
+    errors = number_of_errors(_FILE, database, models=[det_award_1, det_award_2, det_award_3, det_award_4, zips])
+    assert errors == 0
+
+    # invalid 5 digit zip - this should pass but is handled in d41_3
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny10986",
+                                                          place_of_performance_zip4a="12346")
+    det_award_2 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="NA*****",
+                                                          place_of_performance_zip4a='12345')
```

Comment:
Adding to these tests, there should be a 9-digit here with an invalid first 5 to see if it's properly ignored.